x/ibchooks/client/cli/cli_test.go

```diff
@@ -0,0 +1,181 @@
+	genesisState := s.cfg.GenesisState
+
+	s.cfg.NumValidators = 1
+	s.generateAccountsWithKeyrings(2)
+
+	var genBalances []banktypes.Balance
+	for i := range s.accountAddresses {
+		genBalances = append(genBalances, banktypes.Balance{Address: s.accountAddresses[i].String(), Coins: sdk.NewCoins(
+			sdk.NewInt64Coin("nhash", 100_000_000), sdk.NewInt64Coin(s.cfg.BondDenom, 100_000_000),
+		).Sort()})
+	}
+	var bankGenState banktypes.GenesisState
+	bankGenState.Params = banktypes.DefaultParams()
+	bankGenState.Balances = genBalances
+	bankDataBz, err := s.cfg.Codec.MarshalJSON(&bankGenState)
+	s.Require().NoError(err, "should be able to marshal bank genesis state when setting up suite")
+	genesisState[banktypes.ModuleName] = bankDataBz
+
+	var authData authtypes.GenesisState
+	var genAccounts []authtypes.GenesisAccount
+	authData.Params = authtypes.DefaultParams()
+	genAccounts = append(genAccounts, authtypes.NewBaseAccount(s.accountAddresses[0], nil, 3, 0))
+	genAccounts = append(genAccounts, authtypes.NewBaseAccount(s.accountAddresses[1], nil, 4, 0))
+	accounts, err := authtypes.PackAccounts(genAccounts)
+	s.Require().NoError(err, "should be able to pack accounts for genesis state when setting up suite")
+	authData.Accounts = accounts
+	authDataBz, err := s.cfg.Codec.MarshalJSON(&authData)
+	s.Require().NoError(err, "should be able to marshal auth genesis state when setting up suite")
+	genesisState[authtypes.ModuleName] = authDataBz
+
+	ibchooksParams := ibchookstypes.DefaultParams()
+	ibchooksGenesis := &ibchookstypes.GenesisState{Params: ibchooksParams}
+
+	ibchooksDataBz, err := s.cfg.Codec.MarshalJSON(ibchooksGenesis)
+	s.Require().NoError(err, "should be able to marshal ibchooks genesis state when setting up suite")
+	genesisState[ibchookstypes.ModuleName] = ibchooksDataBz
+
+	s.cfg.GenesisState = genesisState
+
+	s.cfg.ChainID = antewrapper.SimAppChainID
+
+	s.testnet, err = network.New(s.T(), s.T().TempDir(), s.cfg)
+	s.Require().NoError(err, "network.New")
+
+	// s.testnet.Validators[0].ClientCtx = s.testnet.Validators[0].ClientCtx.WithKeyringDir(s.keyringDir).WithKeyring(s.keyring)
+	_, err = testutil.WaitForHeight(s.testnet, 6)
+	s.Require().NoError(err, "WaitForHeight")
+}
+
+func (s *IntegrationTestSuite) TearDownSuite() {
+	testutil.Cleanup(s.testnet, s.T())
+}
+
+func (s *IntegrationTestSuite) generateAccountsWithKeyrings(number int) {
+	path := hd.CreateHDPath(118, 0, 0).String()
+	s.keyringDir = s.T().TempDir()
+	kr, err := keyring.New(s.T().Name(), "test", s.keyringDir, nil, s.cfg.Codec)
+	s.Require().NoError(err, "Keyring.New")
+	s.keyring = kr
+	for i := 0; i < number; i++ {
+		keyId := fmt.Sprintf("test_key%v", i)
+		info, _, err := kr.NewMnemonic(keyId, keyring.English, path, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+		s.Require().NoError(err, "Keyring.NewMneomonic")
+		addr, err := info.GetAddress()
+		if err != nil {
+			panic(err)
+		}
+		s.accountAddresses = append(s.accountAddresses, addr)
+	}
+}
```

Comment:
Please use `testutil.GenerateTestKeyring` for this. See `x/attribute/client/cli/cli_test.go` for an example.